### ISSUE TITLE
write-cache mode still READ-ONLY when we changed shard mode to READ-WRITE

### ISSUE THREAD

**Issue by anikeev-yadro** [nspcc-dev/neofs-node#1906]
Related to https://github.com/nspcc-dev/neofs-node/issues/1905

## Expected Behavior
If we change shard mode to READ-WRITE, then write-cache mode should be changed to READ-WRITE too.

## Current Behavior
I changed shard mode to READ-WRITE, but write-cache mode kept in READ-ONLY.

## Steps to Reproduce (for bugs)
1. Set shard mode to READ-WRITE:
```
root@az:/srv/neofs# sudo neofs-cli --endpoint 127.0.0.1:8091 -w /etc/neofs/storage/wallet.json control shards set-mode --all --mode read-write
Enter password >
Shard mode update request successfully sent.
```
2. Try to flush cache:
```
root@az:/srv/neofs# sudo neofs-cli --endpoint 127.0.0.1:8091 -w /etc/neofs/storage/wallet.json control shards flush-cache --id MnaJVa2bAqANyXabcajLWQ,KNqV3sT3nSFQKMan7w54QH
Enter password >
Write-cache has been flushed.
```
3. Actually write-cache still in READ-ONLY:
```
Oct 17 08:49:42 az sudo[1559]:     root : TTY=pts/0 ; PWD=/srv/neofs ; USER=root ; COMMAND=/usr/bin/neofs-cli --endpoint 127.0.0.1:8091 -w /etc/neofs/storage/wallet.json control shards flush-cache --id MnaJVa2bAqANyXabcajLWQ,KNqV3sT3nSFQKMan7w54QH
Oct 17 08:49:45 az neofs-node[644]: 2022-10-17T08:49:45.312Z        error        shard/delete.go:43        can't delete object from write cache        {"shard_id": "MnaJVa2bAqANyXabcajLWQ", "error": "write-cache is in read-only mode"}
```
**Logs:**  [bug_ro_errors.zip](https://github.com/nspcc-dev/neofs-node/files/9798866/bug_ro_errors.zip)

**Versions:**

```
NeoFS Storage node
Version: v0.32.0-121-gade19077
GoVersion: go1.18.4
```
## Your Environment
Server setup and configuration:
cloud, 4 VMs, 4 SN, 4 http qw, 4 s3 gw

Operating System and version (uname -a):
linux vedi 5.10.0-16-amd64 https://github.com/nspcc-dev/neofs-node/issues/1 SMP Debian 5.10.127-1 (2022-06-30) x86_64 GNU/Linux

**Comment by fyrchik** [comment]
We could increase the timeout and have this fixed up until the next similar error occurs. And we cannot pick a single default, because it (`maximum possible timeout`) depends on the timeouts in the configuration of all storage nodes participating in the operation. So 30 can fix this case, but certainly there are others.

I think we can ensure that `timeout` can be picked from the configuration (I believe it already should). This way any particular deployment can override this default. 15 seconds looks like a reasonable default to me.

@cthulhu-rider @realloc @carpawell

**Comment by fyrchik** [comment]
Closed via #1906
